trade-finance-manager-api/src/graphql/reducers/mappings/facilities/mapUkefExposure.js

```diff
@@ -1,48 +1,28 @@
 const { formattedNumber } = require('../../../../utils/number');
-const api = require('../../../../v1/api');
-const { calculateNewFacilityValue, calculateUkefExposure } = require('../../../helpers/amendment.helpers');
 const { CURRENCY } = require('../../../../constants/currency.constant');
+const { findLatestCompletedAmendment } = require('../../../helpers/amendment.helpers');
 
 // maps ukef exposure on original value or latest amended value
-const mapUkefExposure = async (facilityTfm, facility) => {
+const mapUkefExposure = (facilityTfm, facility) => {
   if (facilityTfm) {
     const {
       ukefExposure,
       ukefExposureCalculationTimestamp,
-      exchangeRate,
     } = facilityTfm;
 
     // sets facility exposure values and timestamp
     let formattedUkefExposure = formattedNumber(ukefExposure);
     let ukefExposureCalculationTimestampValue = ukefExposureCalculationTimestamp;
-
-    if (facility?._id) {
-      const { _id } = facility;
-
-      // gets the latest value, amendmentId and currency from latest completed amendment with value change
-      const latestCompletedAmendmentValue = await api.getLatestCompletedValueAmendment(_id);
-
-      // if value exists - else there is no completed amendment with value changed
-      if (latestCompletedAmendmentValue?.value) {
-        const { amendmentId } = latestCompletedAmendmentValue;
-        const { coverPercentage, coveredPercentage } = facility.facilitySnapshot;
-        // gets full amendment
-        const fullAmendment = await api.getAmendmentById(_id, amendmentId);
-
-        const { requireUkefApproval, effectiveDate, bankDecision } = fullAmendment;
-        // value of ukefExposureCalculationTime from automatic amendment submission time or manual amendment bankDecision submission time
-        const ukefExposureTimestamp = requireUkefApproval ? bankDecision.effectiveDate : effectiveDate;
-
-        // BSS is coveredPercentage while GEF is coverPercentage
-        const coverPercentageValue = coverPercentage || coveredPercentage;
```

Comment:
are you sure we don't need this anymore?

Reply:
this is being done in calculateAmendmentExposure in amendment.helpers.js in trade-finance-manager-api\src\v1\helpers\amendment.helpers.js .  It does the same as here but keeps it in the db so can just pull the values